tiledb/api/c_api/filter/test/CMakeLists.txt

```diff
@@ -28,7 +28,12 @@ include(common NO_POLICY_SCOPE)
 
 add_executable(unit_capi_filter EXCLUDE_FROM_ALL)
 find_package(Catch_EP REQUIRED)
-target_link_libraries(unit_capi_filter PUBLIC Catch2::Catch2)
+target_link_libraries(unit_capi_filter PUBLIC Catch2::Catch2WithMain)
+target_include_directories(unit_capi_filter 
```

Comment:
We can't tolerate separate calls to `target_include_directories` every time that Catch is a dependency. The include directories for Catch should come through the Catch library and not be needed separately. Something's wrong if these need to be included everywhere.

Reply:
Tried to remove all unnecessary targeting of include directories that this PR/change-set implemented.